controllers/users.js

```diff
@@ -42,9 +45,14 @@ module.exports.getUser = (req, res) => {
 };
 
 module.exports.createUser = (req, res) => {
-  const { name, about, avatar } = req.body;
-
-  User.create({ name, about, avatar })
+  const {
+    name, about, avatar, email, password,
+  } = req.body;
+  bcrypt.hash(password, 10)
+    .then((hash) => User.create({
+      name, about, avatar, email, password: hash,
+    }))
+    .then((createdUser) => User.findById(createdUser._id))
     .then((user) => res.send({ data: user }))
```

Comment:
Можно лучше: изменить список полей, которые будут отправлены при создании записи.
Насколько я понял, поиск по `id` был использован, потому что `password` всё же содержится в `createdUser`. Это одна из особенностей. Настройка `{ select: false }` работает только при получении записей.

Можно это обойти, например, создав отдельный метод:
```js
userSchema.methods.omitPrivate = function omitPrivate() {
  const obj = this.toObject();
  delete obj.password;
  return obj;
};
```

И далее
 ```suggestion
     .then((createdUser) => res.send({ data: createdUser.omitPrivate() }))
```